### ISSUE TITLE
Triggering on shift-cmd-b

### ISSUE THREAD

**Issue by jmound** [philc/vimium#1362]
Shift-cmd-b is the keystroke for "always show bookmarks bar", it also triggers the Vomnibox / "open bookmark" via the 'b'

I think the Vomnibox shouldn't be triggered on that keystroke. If there is a way to disable that keystroke via Vimium's options > excluded URLs and keys, let me know. 

OS: OSX 10.10.1
Chrome: 39.0.2171.95
Vimium: 1.49

**Comment by rmetzler** [comment]
Yeah, it's annoying to see the Vomnibox whenever I press `cmd+shift+b`. I guess the fix is similar to this story https://medium.com/medium-eng/the-curious-case-of-disappearing-polish-s-fa398313d4df#9108

**Comment by mrmr1993** [comment]
I remember reading somewhere that the command key isn't a modifier in javascript; its only effect is firing keyup/keydown events. Can anybody confirm this (or point to how to detect it being held during other events)?

**Comment by rmetzler** [comment]
command key is signaled via `event.metaKey` in JavaScript. Vimium seems to use it here: https://github.com/philc/vimium/blob/471b6a22c2b0b570696b1506f71c6b7b52c2c56a/content_scripts/vimium_frontend.coffee#L465-L480

**Comment by mrmr1993** [comment]
Can somebody on OS X check what shows up in the background page console when you press `cmd-b` and `cmd-shift-b` on a page? That would give some insight into what is happening.

To show the background page console:
* Go to `chrome://extensions`
* check checkbox next to developer mode
* click "inspect background page" under Vimium
* press `esc` to show the console if it's not already visible

**Comment by rmetzler** [comment]
```
main.js:720 checking keyQueue: [ <m-B> ]
main.js:722 new KeyQueue: 
main.js:720 checking keyQueue: [ b ]
main.js:722 new KeyQueue: 
main.js:717 clearing keyQueue
```

**Comment by mrmr1993** [comment]
Is that just pressing `cmd-shift-b`? If it is, I suspect we need to start checking for `event.meta` in the `keyup` handler, and (for now) just bailing if it's true.

This will still have issues (eg. #1411), but I don't think it would make any situations any worse then they already are.

**Comment by rmetzler** [comment]
Yes, this is only `cmd-shift-b` and I also tried to verify if the second `b` is coming from the `keyup` event or not. It seems like it does all come from `keydown` as I press and hold all the keys for more than a second, but the `[b]` comes up in the log almost immediately.

**Comment by rmetzler** [comment]
I also tried to type other shortcuts to find key combinations with similar problems, but so far haven't identified any other `cmd+shift+char` combination triggering similar behavior.

Then I put a breakpoint in the extension and looked at the call stack for `handleKeyDown`. This seems to go the proper extension messaging route as far as I can tell without having any experience in writing browser extensions.

Too bad, I can't see the vimium content_scripts in the background page. :-(

**Comment by mrmr1993** [comment]
The b should be from the `keypress` handler, setting a breakpoint there will show us sending the message for it (and, notably, ignoring `event.meta`).

**Comment by rmetzler** [comment]
yes, I can confirm. The additional `b` comes from `onKeypress` in `vimium_frontend.coffee`.
I'm not sure how to fix it. Do you want me to just add another check for `metaKey` or should I refactor the modifier related code from `onKeydown` into `KeyboardUtils` and reuse this?

**Comment by mrmr1993** [comment]
Several other questions need answering, so we know we're not breaking things (and maybe even some fixes for other issues may arise):

* Is `event.shiftKey` true for all keydown events where shift was held (eg. typing `B`)?
* Are `event.shiftKey` and `event.metaKey` both true for a printing character that uses shift and command (eg. `cmd-shift-4`)?
* (BONUS: Do we get any extra information to indicate when the keypress event is the result of a key combo (eg. the 'accent key' and then e for é on some European keyboards).)

Basically, I would like to know whether we can read the keys pressed, the character generated, both, or some unhelpful combination of parts from both. This will inform us of whether we can/how we should handle this (and related) case(s).

**Comment by rrsahuone** [comment]
Our detailed movers and packers Information mill using unique kind of containers and wrapping procedure to guarantee 100 % protection of your products. 

<a href="http://www.expert5th.in/packers-and-movers-hyderabad/" title="Packers and Movers in Hyderabad">Packers and Movers in Hyderabad</a>
<a href="http://www.expert5th.in/packers-and-movers-bangalore/" title="Packers and Movers in Bangalore">Packers and Movers in Bangalore</a>
<a href="http://www.expert5th.in/packers-and-movers-mumbai/" title="Packers and Movers in Mumbai">Packers and Movers in Mumbai</a>

**Comment by rrsahuone** [comment]
We are extremely focused and always using innovative strategy to finish our job. 

<a href="http://www.expert5th.in/packers-and-movers-gurgaon/" title="Packers and Movers in Gurgaon">Packers and Movers in Gurgaon</a>
<a href="http://www.expert5th.in/packers-and-movers-delhi/" title="Packers and Movers in Delhi">Packers and Movers in Delhi</a>
<a href="http://www.expert5th.in/packers-and-movers-pune/" title="Packers and Movers in Pune">Packers and Movers in Pune</a>

**Comment by rrsahuone** [comment]
We don’t give even only one chance to our customers that may go against us because we truly comprehend the value of our customers and believe in creating pleased customer base always.

<a href="http://www.expert5th.in/packers-and-movers-chennai/" title="Packers and Movers in Chennai">Packers and Movers in Chennai</a>
<a href="http://www.expert5th.in/packers-and-movers-noida/" title="Packers and Movers in Noida">Packers and Movers in Noida</a>
<a href="http://www.expert5th.in/packers-and-movers-navimumbai/" title="Packers and Movers in Navi Mumbai">Packers and Movers in Navi Mumbai</a>
<a href="http://www.expert5th.in/packers-and-movers-thane/" title="Packers and Movers in Thane">Packers and Movers in Thane</a>
<a href="http://www.expert5th.in/packers-and-movers-ghaziabad/" title="Packers and Movers in Ghaziabad">Packers and Movers in Ghaziabad</a>
<a href="http://www.expert5th.in/packers-and-movers-faridabad/" title="Packers and Movers in Faridabad">Packers and Movers in Faridabad</a>

**Comment by rrsahuone** [comment]
When you are looking for best packer and moving service solutions in Bangalore , you have come to right position. Our solutions are far better than any other moving Company. 

<a href="http://www.expert5th.in/packers-and-movers-bangalore/" title="Packers and Movers Bangalore">Packers and Movers Bangalore</a>
<a href="http://www.expert5th.in/packers-and-movers-pune/" title="Packers and Movers Pune">Packers and Movers Pune</a>
<a href="http://www.expert5th.in/packers-and-movers-mumbai/" title="Packers and Movers Mumbai">Packers and Movers Mumbai</a>

**Comment by deiga** [comment]
Is this still an issue? For me it doesn't occur anymore

**Comment by smblott-github** [comment]
@deiga...  Are you saying you experienced this issue previously, but at some point it resolved itself?  In which case, we'd close this issue, yes?

**Comment by deiga** [comment]
@smblott-github Yeah, I was able to repro the issue earlier. But now I can't get it to bug out anymore. I'd kinda wait for @jmound to tell us if he still experiences the issue before closing

**Comment by mrmr1993** [comment]
The issue seemed to be that `cmd-shift-b` was leaking a `b` keypress event in OS X. If this no longer happens then the issue should be resolved.

**Comment by jmound** [comment]
The issue is resolved. I tested with OSX 10.11.1, Chrome 47.0, Vimium 1.53, `cmd-shift-b`just hides/shows the bookmarks bar and doesn't trigger the Vomnibox.

**Comment by tanermetin** [comment]
I could not figure it out how to map CMD key, what is the correct format?
